hackathon_site/event/test_api.py

```diff
@@ -193,3 +194,84 @@ def test_leave_with_not_empty_team(self):
         self.assertEqual(
             Team.objects.filter(team_code=old_team.team_code).exists(), True
         )
+
+
+class EventTeamListsViewTestCase(SetupUserMixin, APITestCase):
+    def setUp(self):
+        super().setUp()
+        self.team = Team.objects.create()
+        self.team2 = Team.objects.create()
+        self.team3 = Team.objects.create()
+
+        self.view = reverse("api:event:team-list")
+
+    def _login(self):
+        self.client.login(username=self.user.username, password=self.password)
+
+    def _login_permission(self):
```

Comment:
We're going to need this in a lot of tests from now on, so it would be good to add it to our setup user mixin as part of the login function. Could you modify the `_login` method of `SetupUserMixin` to be:

```python
def _login(self, permissions=None):
    if permissions is not None:
        self.user.permissions.add(permissions)

    self.client.login(username=self.user.username, password=self.password)
```

Then to use it, in `setUp` you can set 

```python
self.permissions = Permission.objects.filter(content_type__app_label=...)
```

and call login in your later tests by doing `self._login(self.permissions)`